storage/remote/client.go

```diff
@@ -151,7 +151,17 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query) (*prompb.QueryRe
 	httpReq.Header.Set("User-Agent", userAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Read-Version", "0.1.0")
 
-	if span := opentracing.SpanFromContext(ctx); span != nil {
+	var span opentracing.Span
+	if parentSpan := opentracing.SpanFromContext(ctx); parentSpan != nil {
+		span = opentracing.StartSpan(
+			"Remote Read",
+			opentracing.ChildOf(parentSpan.Context()),
+		)
+		defer func() {
+			if span != nil {
```

Comment:
it's never nil in this position

Reply:
It is nil if the defer is called after we have finished the span already